packages/core/test/persistence.spec.ts

```diff
@@ -0,0 +1,126 @@
+import { expect, describe, it, beforeEach, afterEach } from "vitest";
+import { getDefaultLoggerOptions, pino } from "@walletconnect/logger";
+import { ICore, IStore, CoreTypes } from "@walletconnect/types";
+import {
+  DEFAULT_DB_NAME,
+  MockValue,
+  TEST_CORE_OPTIONS,
+  disconnectSocket,
+  initCore,
+  initStore,
+  restartCore,
+  searchRecords,
+  storeTestValues,
+  throttle,
+} from "./shared";
+import { Core, Store } from "../src";
+import { generateRandomBytes32 } from "@walletconnect/utils";
+
+describe("Persistence", () => {
+  let core: ICore;
+  let store: IStore<string, MockValue>;
+
+  beforeEach(async () => {
+    core = await initCore();
+  });
+
+  afterEach(async () => {
+    await disconnectSocket(core.relayer);
+  });
+
+  it("should persist store values across restarts", async () => {
+    store = await initStore(core);
+    await restartCore();
+    expect(store.getAll()).to.toMatchObject(storeTestValues);
+  });
+
+  it("should persist store values of PAIRS across restarts", async () => {
+    // --- setup ---
+    const coreA = core; // allias for clarity
+    const coreB = new Core(TEST_CORE_OPTIONS);
+    await coreB.start();
+
+    // --- after restart routine ---
+    const afterRestart = async () => {
+      const { uri, topic } = await coreA.pairing.create();
+      let hasDeleted = false;
+      coreA.pairing.events.on("pairing_delete", () => {
+        hasDeleted = true;
+      });
+
+      await coreB.pairing.pair({ uri });
+
+      // pairing was created
+      expect(coreA.pairing.pairings.keys.length).toBe(1);
+      expect(coreB.pairing.pairings.keys.length).toBe(1);
+
+      // topic does not exist in history
+      expect(searchRecords(coreA.history.records, topic)).toBe(false);
+      expect(searchRecords(coreB.history.records, topic)).toBe(false);
+
+      // ensure that keychain is updated
+      expect(coreA.crypto.keychain.keychain.has(topic)).toBe(true);
+      expect(coreB.crypto.keychain.keychain.has(topic)).toBe(true);
+
+      // ensure that expiry is updated
+      expect(coreA.expirer.values.length).toBe(1);
+      expect(coreB.expirer.values.length).toBe(1);
+
+      await coreB.pairing.disconnect({ topic });
+
+      await throttle(5000); // wait for pairing_delete event to fire
```

Comment:
Seems like quite a long wait and kinda hacky, we could make this a common helper in `shared/helpers.ts` instead: https://github.com/WalletConnect/walletconnect-monorepo/blob/v2.0/packages/core/test/pairing.spec.ts#L7

i.e. `await waitForEvent(() => hasDeleted)`, which will eventually time out on the test level anyways if it never occurs.

Can also name it `waitForCondition` or similar if clearer

Reply:
In that case can also remove the assertion here: https://github.com/WalletConnect/walletconnect-monorepo/pull/3087/files#diff-b3dad8c6ed345090bf45b36645624477abe10c6156b947bda81adc4d198485cbR72

Since the promise fulfilling is the same as that `expect` being fulfilled

Reply:
I originally used `waitForEvent` but CI was failing. I thought the cause was a timeout from `waitForEvent` so I removed it
https://github.com/WalletConnect/walletconnect-monorepo/actions/runs/5514937803/job/14930814999

I added it back in 🤞

Reply:
Looks good, might have been a fluke with the runner